### PR TITLE
Update lightning_kokkos_catalyst CUDA target to work with Catalyst

### PR DESCRIPTION
**Pull request by maliasadi**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-73064]

### CODE REVIEW

**Review by AmintorDusko — APPROVED** [review]
Good job! Thank you!

File: pennylane_lightning/core/src/simulators/lightning_kokkos/catalyst/CMakeLists.txt [hunk 1]
```diff
@@ -73,19 +73,30 @@ else()
 
 endif()
 
+if(Kokkos_ENABLE_CUDA)
+    message(STATUS "Kokkos_ENABLE_CUDA is ON")
+
+    # KOKKOS_CUDA_OPTIONS:
+    # - disable_malloc_async: Disable asynchronous memory allocation to ensure
+    #   that memory operations complete before proceeding.
+    # - enable_lambda: Allow the use of lambda expressions in parallel
```
Comment (multiphaseCFD) [inline pennylane_lightning/core/src/simulators/lightning_kokkos/catalyst/CMakeLists.txt:82]: Nice one! This flag was a blocker for using lambda functor when I was developing kernels for `lightning.kokkos`.

Comment (multiphaseCFD) [reply pennylane_lightning/core/src/simulators/lightning_kokkos/catalyst/CMakeLists.txt:82]: would `enable_lambda` work for older nvcc version? Do we have to add nvcc version requirements as well? Same thing for `enable_constexpr`. `nvcc-7.5` and later?

**Review by multiphaseCFD — COMMENTED** [review]
Thanks @maliasadi , awesome work! Just a couples of q.

File: pennylane_lightning/core/src/simulators/lightning_kokkos/catalyst/CMakeLists.txt (hunk 1 above)
Comment (maliasadi) [reply pennylane_lightning/core/src/simulators/lightning_kokkos/catalyst/CMakeLists.txt:82]: Good point! `enable_lambda` is supported for NVCC 7+ and `enable_constexpr` for 7.5+. I'll update the code to address this. Thanks!

**Review by multiphaseCFD — APPROVED** [review]
Nice work! Thanks @maliasadi !

**Pull request merged**